src/polymer/polymer-element.ts

```diff
@@ -204,39 +212,53 @@ function resolveElement(
   return clone;
 }
 
-function getFlattenedAndResolvedBehaviors(
+export function getFlattenedAndResolvedBehaviors(
     behaviorAssignments: ScannedBehaviorAssignment[], document: Document) {
   const resolvedBehaviors = new Set<Behavior>();
-  _getFlattenedAndResolvedBehaviors(
+  const warnings = _getFlattenedAndResolvedBehaviors(
       behaviorAssignments, document, resolvedBehaviors);
-  return resolvedBehaviors;
+  return {resolvedBehaviors, warnings};
 }
 
 function _getFlattenedAndResolvedBehaviors(
     behaviorAssignments: ScannedBehaviorAssignment[], document: Document,
     resolvedBehaviors: Set<Behavior>) {
+  const warnings: Warning[] = [];
   for (const behavior of behaviorAssignments) {
-    const foundBehavior = document.getOnlyAtId('behavior', behavior.name);
-    if (!foundBehavior) {
-      if (behavior.sourceRange.file === document.url) {
-        document.warnings.push({
-          message: `Unable to resolve behavior ` +
-              `\`${behavior.name}\`. Did you import it? Is it annotated with ` +
-              `@polymerBehavior?`,
-          severity: Severity.ERROR,
-          code: 'unknown-polymer-behavior',
-          sourceRange: behavior.sourceRange
-        });
-      }
+    const foundBehaviors = document.getById('behavior', behavior.name);
+    if (foundBehaviors.size === 0) {
+      warnings.push({
+        message: `Unable to resolve behavior ` +
+            `\`${behavior.name}\`. Did you import it? Is it annotated with ` +
+            `@polymerBehavior?`,
+        severity: Severity.ERROR,
+        code: 'unknown-polymer-behavior',
+        sourceRange: behavior.sourceRange
+      });
+      // Skip processing this behavior.
       continue;
     }
+    if (foundBehaviors.size > 1) {
+      warnings.push({
+        message: `Found more than one behavior named ${behavior.name}.`,
+        severity: Severity.WARNING,
+        code: 'multiple-polymer-behaviors',
+        sourceRange: behavior.sourceRange
+      });
+      // Don't skip processing this behavior, just take the most recently
+      // declared instance.
+    }
+    const foundBehavior = Array.from(foundBehaviors)[foundBehaviors.size - 1];
```

Comment:
That gets you the first one, but we actually want the last one, as last write wins.

(which itself isn't quite right because you can get different orderings of the same imports based on the shape of the total import graph and the starting node, but this is at least closer to correct)

(ok there's even more caveats because it could be that you want one in the _middle_ of the list of results because at runtime we look up the identifier at the time of the behavior reference… but we definitely don't want to bother with that (even though at one point in this code we totally did, the caching story is a nightmare))